src/lib/shared/content-processor/markdown-process.js

```diff
@@ -121,15 +124,19 @@ const markedProcessorFactory = ({
       try {
         if (!imageFiles) {
           // For HTML render
-          const imageName = token.href.split('/').pop();
-          const imageExt = window.contentConfig.imageFileName?.[imageName] || imageName;
+          const imageId = token.href.split('/').pop();
+          const imageExt = window.contentConfig.images[imageId];
           return `<img src="./images/${imageExt}" alt="${token.text}">`;
         }
         // For editor preview
         const imageFile = imageFiles[token.href];
-        if (!imageFile) return `<img src="${token.href}" alt="${token.text}">`;
-        const blobUrl = URL.createObjectURL(imageFile);
-        return `<img src="${blobUrl}" alt="${token.text}">`;
+        
+        if (!blobUrlCache.has(token.href)) {
+          const blobUrl = URL.createObjectURL(imageFile);
+          blobUrlCache.set(token.href, blobUrl);
+        }
```

Comment:
I like the idea of caching the blob by href. Could we create a helper function that returns the blob URL and manages the cache internally?

Reply:
It should be easier to use. Perhaps you could implement it in closure if you prefer, as I do.